collection/delete.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2015-2018 Kuzzle
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collection
+
+import (
+	"github.com/kuzzleio/sdk-go/types"
+)
+
+// Delete a collection
+func (dc *Collection) Delete(index string, collection string, options types.QueryOptions) error {
+	if index == "" {
+		return types.NewError("Collection.Delete: index required", 400)
+	}
+
+	if collection == "" {
+		return types.NewError("Collection.Delete: collection required", 400)
+	}
```

Comment:
You should let Kuzzle handle the errors if an index or collection name is empty
```suggestion
```

Reply:
All the other controller actions have this kind of double check on input args. I think we should create an issue for those changes. What do you think @Aschen?

Reply:
Yes we should remove every double check from the Go SDK 👍